src/api/tasksService.js

```diff
@@ -0,0 +1,350 @@
+//importing files...
+import api from "./axiosConfig";
+
+//UTILITY FUNCTION
+/**
+ * getNextTaskId - Asynchronous function that determines the ID of the next task
+ * @param {String} endpoint: the respective endpoint to get the task array from
+ * @returns {Number} returns the ID that should be used for the next task to be created
+ */
+const getNextTaskId = async (endpoint) => {
+  try {
+    const res = await api.get(endpoint);
+    const items = res.data;
+    const validIds = items.map((item) => Number(item.id));
+    const maxId = validIds.length > 0 ? Math.max(...validIds) : 0;
+    return maxId + 1;
+  } catch (err) {
+    console.error("Error setting next ID for item:", err);
+    throw err;
+  }
+};
+
+/**
+ * getNextSubtaskOrNoteId - Asynchronous function that determines the ID of the next subtask or note
+ * @param {Array} items: Array of subtasks or notes
+ * @returns {Number} returns the ID that should be used for the next subtask or note to be created
+ */
+const getNextSubtaskOrNoteId = async (items) => {
+  try {
+    const maxId =
+      items.length > 0 ? Math.max(...items.map((item) => item.id)) : 0;
+    return maxId + 1;
+  } catch (err) {
+    console.error("Error setting next ID for item:", err);
+    throw err;
+  }
+};
+
+/**
+ * removeID - removes any ID passed with the task or subtask object
+ * @param {object} data: task or subtask to strip ID off
+ * @returns {object} returns a task or subtask without the ID
+ */
+const removeId = (data) => {
+  const { id, ...rest } = data;
+
+  return rest;
+};
+
+//GET ENDPOINTS
+/**
+ * getAllTasks - Asynchronous function that gets all tasks in the db
+ * @returns {Promise<Array>} All tasks created
+ */
+export const getAllTasks = async () => {
+  try {
+    const res = await api.get("/tasks");
+    return res.data;
```

Comment:
Collaborate with @Haydox01 if there is an option to exclude some keys so we dont get a heavy payload from the server.